trigger.js

```diff
@@ -215,39 +224,76 @@ class TriggerHappy {
     _doCaptureTriggers(token, scene, update) {
         // Get all trigger tokens in scene
         let targets = this._getTokensFromTriggers(canvas.tokens.placeables, this.triggers, 'capture');
-        if (!targets) return true;
+        if (!targets) return;
 
         const finalX = update.x || token.x;
         const finalY = update.y || token.y;
         // need to calculate this by hand since token is just token data
-        const tw = token.width * canvas.scene.data.grid / 2;
-        const th = token.height * canvas.scene.data.grid / 2;
-        const motion = new Ray({x: token.x + tw, y: token.y + th}, {x: finalX + tw, y: finalY + th});
+        const tkw = token.width * canvas.scene.data.grid / 2;
+        const tkh = token.height * canvas.scene.data.grid / 2;
+
+        const motion = new Ray({x: token.x + tkw, y: token.y  + tkh}, {x: finalX + tkw, y: finalY  + tkh});
 
         // don't trigger on tokens that are already captured
-        targets = targets.filter(target => token.x + tw !== target.center.x || token.y + th !== target.center.y)
-        
+        targets = targets.filter(target =>  !this._tokenContains(target, {x: token.x + tkw, y: token.y  + tkh}));
+
         // sort list by distance from start token position
         targets.sort((a , b) => targets.sort((a, b) => Math.hypot(token.x - a.x, token.y - a.y) - Math.hypot(token.x - b.x, token.y - b.y)))
         
         for (let target of targets) {
-            // test motion vs token diagonals
-            if (motion.intersectSegment([target.x, target.y, target.x + target.w, target.y + target.h])
-            || motion.intersectSegment([target.x, target.y + target.h, target.x + target.w, target.y])) {
-                update.x = target.center.x - tw;
-                update.y = target.center.y - th;
-                return true;
+            const tx = target.x;
+            const ty = target.y;
+            const tw = target.w;
+            const th = target.h;
+            const tgw = target.data.width; // target token width in grid units
+            const tgh = target.data.height; // target token height in grid units
+                // test motion vs token diagonals
+            if (tgw > 1 && tgh > 1 && tgw * tgh > 4) {
+                // big token so do boundary lines
+                var intersects = ( motion.intersectSegment([tx,      ty,      tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx,      ty + th])
+                                || motion.intersectSegment([tx,      ty,      tx,      ty + th]))
+            } else  // just check the diagonals
+                var intersects = (motion.intersectSegment([tx,       ty,      tx + tw, ty + th])
+                               || motion.intersectSegment([tx,       ty + th, tx + tw, ty     ]));

```

Comment:
oh yeah, I like the way you displayed them! 👍 
Please use `{ }` however for an non-onliner if (even if it's a one-statement if). Especially in this case, if you use `{ }` for the block of the `if`, then use it also for the `else` block, even if that one is a one-liner. It's otherwise confusing.